<?xml version="1.0"?>
<!--
     !!!EXPERIMENTAL!!!
     !!! ONLY BINARY COMMAND SUPPORT !!!
     **********************************************
     Lanch File for setting new ip address
     **********************************************
    This is an experimental launch file to set a new ip address in the scanner.
    Maybe one should add a EEPROM-write command. But in our tests this was not necessary.
-->

<launch>
  <arg name="hostname" default="192.168.0.1" />
<arg name="new_IP" default="192.168.0.1" />
  <arg name="cloud_topic" default="cloud" />
  <arg name="frame_id" default="cloud" />
  <node name="sick_tim_5xx" pkg="sick_scan" type="sick_generic_caller"  respawn="false" output="screen">
  <!--<param name="robot_description" command="$(find xacro)/xacro.py '$(find sick_scan)/urdf/example.urdf.xacro'" />-->
    <param name="scanner_type" type="string" value="sick_tim_5xx" />
    <!-- -135° -->
    <param name="min_ang" type="double" value="-2.35619449" />
    <!-- 135° -->
    <param name="max_ang" type="double" value="2.35619449" />
    <param name="use_binary_protocol" type="bool" value="False" />
    <param name="range_max" type="double" value="100.0" />
    <param name="intensity" type="bool" value="True" />
    <param name="hostname" type="string" value="$(arg hostname)" />
    <param name="cloud_topic" type="string" value="$(arg cloud_topic)"/>
    <param name="frame_id" type="str" value="$(arg frame_id)" />
    <param name="port" type="string" value="2112" />
    <param name="timelimit" type="int" value="5" />
    <param name="new_IP_address" type="string" value="$(arg new_IP)"/>
  </node>
</launch>


